kategory-docs/docs/docs/typeclasses/eq/README.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: docs
+title: Eq
+permalink: /docs/typeclasses/eq/
+---
+
+## Eq
+
+The `Eq` typeclass abstracts the ability to compare two instances of any object.
+Depending on your needs this comparison can be structural -the content of the object-, referential -the memory address of the object-, based on an identity -like an Id fields-, or any combination of the above.
+
+It can be considered the typeclass equivatent of Java's `Object#equals`.
+
+### Main Combinators
+
+#### eqv
+
+Compares two instances of F and returns true if they're considered equal for this instance.
+It is the opposite comparison of `neqv`.
+
+`fun eqv(a: F, b: F): Boolean`
+
+```kotlin:ank
+Option.pure(1) // Option(1)
+```
+
+#### neqv
+
+Compares two instances of F and returns true if they're not considered equal for this instance.
+It is the opposite comparison of `eqv`.
+
+`fun neqv(a: F, b: F): Boolean`
+
+```kotlin:ank
+Option.pure(1) // Option(1)
+```
+
+### Syntax
+
+#### HK<F, A>#eqv
+
+Lift a value into the computational context of a type constructor
+
+```kotlin:ank
+1.eqv(2)
+```
+
+#### HK<F, A>#neqv
+
+Apply a function inside the type constructor's context
+
+```kotlin:ank
+1.neqv(2)
+```
+
+### Laws
+
+Kategory provides [`EqLaws`](/docs/typeclasses/laws#eqlaws) in the form of test cases for internal verification of lawful instances and third party apps creating their own `Eq` instances.
+
+#### Creating your own `Eq` instances
+
+Eq provides one special instance that can be potentially applicable to most datatypes.
+It uses kotlin's == comparison to compare any two instances.
+Note that this instance will fail on many all datatypes that contain a property or field that doesn't implement structural equality, i.e. functions, typeclasses, non-data classes
+
+```kotlin:ank
+Eq.any().eqv(1, 2)
+```
+
+```kotlin:ank
+Eq.any().eqv(Either.right(1), Either.pure(1))
+```
+
+```kotlin:ank
+Eq.any().eqv(IO{ 1 }, IO{ 1 })
+```
+
+See [Deriving and creating custom typeclass] to provide your own `Eq` instances for custom datatypes.
+
+### Data types
+
+The following datatypes in Kategory provide instances that adhere to the `Eq` typeclass.
```

Comment:
We can remove this section and use:
> ```kotlin:ank
> import kategory.debug.*
> 
> showInstances<EqHK, Unit>()
> ```

Reply:
Problem is, there are now just two instances: Free and FreeApplicative and none have a global instance object. We're using Eq.any() and test-specific instances everywhere in the project.

Reply:
Should we have a special case for Eq where global lookup always returns Eq.Any()?

Reply:
many more instances are possible but we just have not implemented them yet. Option, Try, Id, Either, ... can all have default `Eq` instances automatically derived. I would leave it as is even if it only shows 2 now because next time the docs compile and those are there it will display them

Reply:
I've opened a ticket for them: https://github.com/kategory/kategory/issues/383